udata/search/__init__.py

```diff
@@ -195,32 +186,38 @@ def facets_kwargs(adapter, facets=None):
         ]
 
 
-def query(model, **kwargs):
-    adapter = adapter_catalog[model]
-    facets = kwargs.pop('facets', None)
-    facets = facets_kwargs(adapter, facets)
-    if isinstance(facets, basestring):
-        facets = [facets]
+def search_for(model, **params):
+    is_adapter = issubclass(model, ModelSearchAdapter)
+    adapter = model if is_adapter else adapter_catalog[model]
+    facets = facets_for(adapter, params.pop('facets', None))
     facet_search = adapter.facet_search(*facets)
-    q = kwargs.pop('q', '')
-    s = facet_search(q, kwargs)
-    r = s.execute()
-    from beeprint import pp
-    pp(r.facets)
-    return SearchResult(s, r)
+    q = params.pop('q', '')
+    return facet_search(q, params)
+
+
+def query(model, **kwargs):
+    params = multi_to_dict(request.args)
+    search = search_for(model, **params)
+    result = search.execute()
+    return SearchResult(search, result)
 
 
 def iter(*adapters, **kwargs):
     return SearchQuery(*adapters, **kwargs).iter()
 
 
-def multisearch(*queries):
+def multisearch(*models, **params):
     ms = MultiSearch(using=es.client, index=es.index_name)
-    for query in queries:
-        qs = query.build_query()
-        ms = ms.add(qs)
+    queries = []
+    for model in models:
+        s = search_for(model, **params)
+        ms = ms.add(s._s)
+        queries.append(s)
     responses = ms.execute()
-    return responses
+    return [
+        SearchResult(query, response)
+        for response, query in zip(responses, queries)
```

Comment:
What about swapping values within the zip to be consistent with the order of params for SearchResult?